[ADD] inventory: stock report (#9289)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/reporting/stock.rst

============
Stock report
============

Use the stock report in Odoo *Inventory* for a detailed list of all stored products, including those
reserved, purchased and in transit, as well as those delivered to customers.

> .. note::
>    The reporting feature only accessible to users with :doc:`admin access
>    <../../../../general/users/access_rights>`.

Reviewer: ```suggestion
.. note::
   The reporting feature is only accessible to users with :doc:`admin access
   <../../../../general/users/access_rights>`.
```

To access the stock report, go to :menuselection:`Inventory app --> Reporting --> Stock`.

.. image:: stock/stock-report.png
   :align: center
   :alt: Show the stock report, accessible by going to Inventory > Reporting > Stock.

Navigate the stock report
=========================

> On the stock report, the left sidebar includes :guilabel:`Warehouses` to filter products by specific
> warehouses and :guilabel:`Category`, to show products within a selected product category.

Reviewer: ```suggestion
On the stock report, the left sidebar includes several groupings to narrow down what is being shown. The default groupigs are :guilabel:`Warehouses`,  which filters products by specific
warehouses, and :guilabel:`Category`, which shows products within a selected product category.
```

I would also add a note that the Wareouses only shows up if in a ulti=warehouse nevironment. In Runbot I only see the categories.

In the report itself, the columns represent:

- :guilabel:`Product`: name of the product.

> - :guilabel:`Unit Cost`: average inventory valuation per unit, adjusted based on purchasing costs.

Reviewer: would a link to the documentation on valuation help? I know there is aFIFO, LIFO, AVCO, etc- and it cna affect the cost? Is that relevant to mention or link to?

Author: Added the costing methods doc! thank you for the suggestion :smile:

- :guilabel:`Total Value`: Total inventory valuation of the product, calculated by multiplying unit
  cost by on-hand quantity.

.. seealso::

>    :doc:`../inventory_valuation/using_inventory_valuation`

Reviewer: I think I am confused- is the cost, the cost to purchase or manufacture? And then the value is different? I just think they can be explained a touch further since I am not 100% sure the difference between the two.

Author: Added some more details! thank you (:

- :guilabel:`On Hand`: current quantity of products. Click the :icon:`fa-pencil`
  :guilabel:`(pencil)` icon to :doc:`modify the on-hand quantity
  <../inventory_management/count_products>`.

> - :guilabel:`Free to Use`: on-hand quantity that are not reserved for delivery or manufacturing
>   orders.

Reviewer: ```suggestion
- :guilabel:`Free to Use`: on-hand quantity that are **not** reserved for delivery or manufacturing
  orders, and are available to sell or use.
```

> - :guilabel:`Incoming`: items expected to arrive at the warehouse.

Reviewer: Is this based on the number of approved/confirmed PO's? I might expand this so we know where this number is coming from. I assume it's fro the PO- but is it nly ocunting confirmed, any on order, quuotes requested, etc?

- :guilabel:`Outgoing`: items expected to leave the warehouse or be consumed in manufacturing

>   orders.

Reviewer: again- expected ot leave on coonfirmes sales orders? I'd just be super specific.

Author: Yes, thank you! Adding the clarification helps

Click the buttons to the right of each row item to access additional information:

- :guilabel:`History`: stock move history of the product, displaying information about the quantity
  and description of why the product was moved from one location to another.
- :guilabel:`Replenishment`: reordering rules to procure the product
- :guilabel:`Locations`: break down of on-hand quantity at multiple storage locations
- :guilabel:`Forecast`: forecasted report to view on-hand, incoming, and outgoing quantities, and
  confirmed purchase, sales, or manufacturing orders

> 

Reviewer: in general- I noticed the buttons appear regardless if there is info to show or not. Example, when I clicked 'replenishment' it took me to a screen that said 'there are no re-ordering rules for this product'. So, when you click it, does it hsow the rule,s or does it actually reorder? I might oduble check for that, and let people know the buttons appear regardless if there is info?

Author: Suggestion implemented (:

Search options
--------------

.. tabs::

>    .. tabs:: Filters

Reviewer: you can also reference a generic reporting doc- I believe there is one that explains the search/filter/etc?

      The :guilabel:`Filters` section allows users to search among pre-made and custom filters to
      find specific stock records.

      - :guilabel:`Published`: display products published on the website. Only available with the
        *Website* app installed.
      - :guilabel:`Available in POS`: display products available through the *Point of Sale*
        app.
      - :guilabel:`Available in Self`: display products available in self order through the *Point
        of Sale* app.
      - :guilabel:`Not available in Self`: display products available in *PoS*, but not available in
        self order.

      .. seealso::

>          `Configure PoS products <https://youtu.be/REbA3TBhFa4>`_

Reviewer: ooh i like the youtube link!! GOOD IDEA!!!

      - :guilabel:`Can be Sold`: display products that can be sold to customers. Appears in the

>         search because the *Can be Sold* checkbox is ticked on the product form.

Reviewer: could possibly link to the product form part of a doc for adding a product?

>       - :guilabel:`Can be Purchased`: display products that can be bought from vendors.

Reviewer: also labeled as such on the product form?
